lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""This library facilitates the prometheus_remote_write interface.
+
+For the provider charm no communication is necesssary. For the consumer charm,
+instantite a PrometheusRemoteWriteConsumer in the __init__ method of your charm
+class:
+
+```
+from charms.prometheus_k8s.v0.prometheus_remote_write import PrometheusRemoteWriteConsumer
+
+def __init__(self, *args):
+    ...
+    self.remote_write = PrometheusRemoteWriteConsumer(self, "remote-write")
+    ...
+```
+
+Then access the endpoints with:
+`self.remote_write.endpoints`
+or access a prometheus style config object with:
+`self.remote_write.configs`
+"""
+
+from typing import List, Optional, Union
+
+from ops.charm import CharmBase
+from ops.framework import Object
+from ops.model import BlockedStatus
+
+
+class PrometheusRemoteWriteConsumer(Object):
+    """A prometheus remote write consumer."""
+
+    def __init__(self, charm: CharmBase, relation_name: str):
+        """A prometheus remote write consumer.
+
+        Args:
+            charm: The charm object that instantiated this class.
+            relation_name: The relation name as defined in metadata.yaml.
+        """
+        super().__init__(charm, relation_name)
+        self._relation_name = relation_name
+
+    @property
+    def endpoints(self) -> List[str]:
+        """A list of remote write endpoints.
+
+        Returns:
+            A list of remote write endpoints.
+        """
+        endpoints = []
+        for relation in self.model.relations[self._relation_name]:
+            for unit in relation.units:
+                # If external-address is provided use that, else use ingress-address
+                if (address := relation.data[unit].get("address")) is None:
+                    continue
+                if (port := relation.data[unit].get("port")) is None:
+                    continue
+                endpoints.append(f"http://{address}:{port}/api/v1/write")
+        return endpoints
+
+    @property
+    def configs(self) -> list:
+        """A config object ready to be dropped in to a prometheus config file.
```

Comment:
Is this config only used for "prometheus config file" ? Could it also be used for "grafana agent config file" ? Even though the two may be exactly the same I wonder if it should be made explicit which applications this config file is compatible with.

Reply:
In the grafana agent config file, you include a literal section of a prometheus config file. This is basically saying "can be put in a config file as defined by Prometheus"

Reply:
That sounds fine I will change it.